packages/api/tests/fieldCrop.test.js

```diff
@@ -0,0 +1,562 @@
+      let fieldCrop = fakeFieldCrop();
+      delete fieldCrop.area_used;
+      postFieldCropRequest(fieldCrop, (err, res) => {
+        console.log(fieldCrop,res.error);
+        expect(res.status).toBe(400);
+        expect(JSON.parse(res.error.text).error.data.area_used[0].keyword).toBe("required");
+        done()
+      })
+    });
+
+    test('should return 400 status if fieldCrop is posted w/o estimated_revenue', (done) => {
+      let fieldCrop = fakeFieldCrop();
+      delete fieldCrop.estimated_revenue;
+      postFieldCropRequest(fieldCrop, (err, res) => {
+        console.log(fieldCrop,res.error);
+        expect(res.status).toBe(400);
+        expect(JSON.parse(res.error.text).error.data.estimated_revenue[0].keyword).toBe("required");
+        done()
+      })
+    });
+
+    test('should return 400 status if fieldCrop is posted w/o estimated_production', (done) => {
+      let fieldCrop = fakeFieldCrop();
+      delete fieldCrop.estimated_production;
+      postFieldCropRequest(fieldCrop, (err, res) => {
+        console.log(fieldCrop,res.error);
+        expect(res.status).toBe(400);
+        expect(JSON.parse(res.error.text).error.data.estimated_production[0].keyword).toBe("required");
+        done()
+      })
+    });
+
+    test('should return 400 status if fieldCrop is posted w/ area > field.area', (done) => {
+      let fieldCrop = fakeFieldCrop();
+      fieldCrop.area_used = field.area + 1;
+      fieldCrop.estimated_production = 1;
+      fieldCrop.estimated_revenue = 1;
+      postFieldCropRequest(fieldCrop, (err, res) => {
+        console.log(fieldCrop,res.error);
+        expect(res.status).toBe(400);
+        expect(res.error.text).toBe('Area needed is greater than the field\'s area');
+        done()
+      })
+    });
+
+    test('should return 400 status if fieldCrop is posted w/ area < 0', (done) => {
+      let fieldCrop = fakeFieldCrop();
+      fieldCrop.area_used = -1;
+      fieldCrop.estimated_production = 1;
+      fieldCrop.estimated_revenue = 1;
+      postFieldCropRequest(fieldCrop, (err, res) => {
+        console.log(fieldCrop,res.error);
+        expect(res.status).toBe(400);
+        expect(JSON.parse(res.error.text).error.data.area_used[0].message).toBe("should be >= 0");
+        done()
+      })
+    });
+
+    test('Should post then get a valid fieldcrop (bed size and percentage)', (done) => {
+      let fieldCrop = fakeFieldCrop();
+      fieldCrop.estimated_revenue = 1;
+      fieldCrop.area_used = field.area * 0.25;
+      fieldCrop.estimated_production = 1;
+      postFieldCropRequest(fieldCrop, (err, res) => {
+        console.log(fieldCrop,res.error);
+        expect(res.status).toBe(201);
+        getRequest(`/field_crop/farm/${farm.farm_id}`,(err, res)=>{
+          expect(res.status).toBe(200);
+          expect(res.body[0].field_id).toBe(fieldCrop.field_id);
+          expect(res.body[0].crop_id).toBe(fieldCrop.crop_id);
+          done();
+        })
```

Comment:
let's not do that here, instead of doing a get request, query the db directly. This is because, if the GET endpoint starts failing, we want to know it only on the GET endpoint test, if we have this here, then we will have multiple test failing, which may seem like a bigger problem than it actually is.